wallet/src/de/schildbach/wallet/WalletApplication.java

```diff
@@ -995,15 +1008,8 @@ public void finalizeWipe() {
             walletBackupFile.delete();
         }
 
-        Executors.newSingleThreadExecutor().execute(new Runnable() {
-            @Override
-            public void run() {
-                PlatformRepo.getInstance().clearDatabase(true);
-            }
-        });
-
         // clear data on wallet reset
-        transactionMetadataProvider.clear();
+        WalletApplicationExt.INSTANCE.clearDatabases(this, true);
```

Comment:
Here we are removing several different run and forget methods and replacing it with a single run and forget method.